File: pactfix-py/dockerfiles/Dockerfile.php
FROM php:8.3-cli
WORKDIR /app
COPY --from=composer:latest /usr/bin/composer /usr/bin/composer
COPY composer.* ./
RUN composer install 2>/dev/null || true
COPY . .
CMD ["php", "-S", "0.0.0.0:8080", "-t", "public", "||", "php", "index.php"]

File: pactfix-py/dockerfiles/Dockerfile.java
FROM eclipse-temurin:21-jdk-jammy
WORKDIR /app
COPY . .
RUN if [ -f "pom.xml" ]; then ./mvnw package -DskipTests 2>/dev/null || mvn package -DskipTests; fi
RUN if [ -f "build.gradle" ]; then ./gradlew build -x test 2>/dev/null || gradle build -x test; fi
CMD ["java", "-jar", "target/*.jar", "||", "java", "Main"]
